app/domain/operations/hbx_enrollments/drop_enrollment_members.rb

```diff
@@ -11,51 +11,101 @@ class DropEnrollmentMembers
 
         Success(params)
       end
 
       def drop_enrollment_members(params)
-        terminate_date = Date.strptime(params[:options]["termination_date_#{params[:hbx_enrollment].id}"], "%m/%d/%Y")
+        termination_date = Date.strptime(params[:options]["termination_date_#{params[:hbx_enrollment].id}"], "%m/%d/%Y")
         dropped_enr_members = params[:options].select{|string| string.include?("terminate_member")}.values
         all_enr_members = params[:hbx_enrollment].hbx_enrollment_members
-        eligible_members = all_enr_members.reject{ |member| dropped_enr_members.include?(member.id.to_s) }
-
-        reinstatement = Enrollments::Replicator::Reinstatement.new(params[:hbx_enrollment], terminate_date + 1.day, 0, eligible_members).build
-        reinstatement.save!
+        non_eligible_members = all_enr_members.select{ |member| dropped_enr_members.include?(member.id.to_s) }
+        new_effective_date = (termination_date > params[:hbx_enrollment].effective_on) ? termination_date + 1.day : params[:hbx_enrollment].effective_on
+        return Failure('Select termination date that would result member drop in present calender year.') unless new_effective_date.year == termination_date.year
 
-        get_household_applied_aptc(reinstatement, terminate_date + 1.day) if params[:hbx_enrollment].applied_aptc_amount > 0
+        custom_params = custom_params(params[:hbx_enrollment], new_effective_date)
+        clone_enrollment = Operations::HbxEnrollments::Clone.new.call({hbx_enrollment: params[:hbx_enrollment], effective_on: new_effective_date, options: custom_params})
+        return clone_enrollment if clone_enrollment.failure?
 
-        reinstatement.force_select_coverage! if reinstatement.may_reinstate_coverage?
-        reinstatement.begin_coverage! if reinstatement.may_begin_coverage? && reinstatement.effective_on <= TimeKeeper.date_of_record
+        new_enrollment = clone_enrollment.value!
+        new_enrollment.hbx_enrollment_members.delete_if {|mem| non_eligible_members.pluck(:applicant_id).include?(mem.applicant_id)}
+        set_product_id(params[:hbx_enrollment], new_enrollment)
+        new_enrollment.check_for_subscriber
+        new_enrollment.generate_hbx_signature
+        new_enrollment.select_coverage! if new_enrollment.may_select_coverage?
+        update_member_effective_dates(new_enrollment)
+        return Failure('Failed to save dropped member enrollment') unless new_enrollment.persisted?
 
-        params[:hbx_enrollment].update_attributes!(terminated_on: terminate_date)
+        get_household_applied_aptc(new_enrollment, new_effective_date) if params[:hbx_enrollment].applied_aptc_amount > 0
+        terminate_base_enrollment(params[:hbx_enrollment], new_enrollment, termination_date)
 
```

Comment:
EDI events are not handled correctly [here](https://github.com/ideacrew/enroll/blob/trunk/app/models/forms/bulk_actions_for_admin.rb#L71 ) When Transmit to Carrier? unchecked.

https://github.com/ideacrew/enroll/blob/trunk/app/models/forms/bulk_actions_for_admin.rb#L71 revert edi logic from here handle as below in dep drop service

When unchecked no EDI goes out for both base enrollment termination and dep drop enrollment purchase.
Current logic when unchecked sends EDI for new purchase.

<img width="1170" alt="Screen Shot 2022-04-20 at 12 25 11 AM" src="https://user-images.githubusercontent.com/14282458/164150226-b83f7871-7563-4add-844b-c41fb2dd8b01.png">


you can have method smthg like this after base enrollment terminated successfully 
```
 def notify_trading_partner
    notify = params[:options].present? && (params[:options][:notify].is_a?(Boolean) && params[:options]. 
                  [:notify].to_s == "false") ? params[:options][:notify] : true
        base_enrollment.notify_enrollment_cancel_or_termination_event(notify)
        new_enrollment.notify_of_coverage_start(notify)
      end
```